### ISSUE TITLE
Couldn't unzip, the file is corrupt

### ISSUE THREAD

**Issue by austonpramodh** [acecilia/OpenWRTInvasion#82]
error:
```
start uploading config file...
{"code":1629,"msg":"Couldn't unzip, the file is corrupt"}
```

Firmware: MiWiFi Stable 3.0.16
I tried the exploit script v0.0.1 and v0.0.6. Both have same errors.

I am executing these scripts on WSL2.

V0.0.1
![image](https://user-images.githubusercontent.com/5049050/117114157-206c9f80-ada9-11eb-84ce-f7f7f20ed21a.png)

V0.0.6
![image](https://user-images.githubusercontent.com/5049050/117114209-2bbfcb00-ada9-11eb-8108-761faa82b193.png)

**Comment by acecilia** [comment]
Cant help much. This issue is usually due to a complex network configuration: try to simplify it. You can see other similar reported issues in this repository.

**Comment by austonpramodh** [comment]
Yes, I understand now. I was able to successfully install openwrt. Thank you.

The issue is caused because the port 4444 exposed on PC might not be contactable from the router.

For the users experiencing the same issue:
When you start netcat, your pc starts listening at 4444. Then the exploit prepares a tar file with script and uploads it to the router. When the router executes the script, it tries connecting to the PC with the IP address provided to the script. 

In my case, maybe the router wasn't able to connect to the netcat session. So I ran same thing on raspberry pi and it worked. May be because it didn't have any firewall configured.  Also, you might wonder the stok doesn't work as it depends on the clients IP. I had SOCKS port forwarded ([how-to-setup-ssh-socks-tunnel-for-private-browsing](https://linuxize.com/post/how-to-setup-ssh-socks-tunnel-for-private-browsing/)) from raspberry pi to firefox, logged into the router and got the stok.
